File: app/src/main/java/org/flyontime/android/dagger/ApplicationComponent.java
package org.flyontime.android.dagger;

import org.flyontime.android.FlyOnTimeApplication;
import org.flyontime.android.ui.presenter.HomePresenter;

import javax.inject.Singleton;

import dagger.Component;

/**
 * Created by JW on 27.04.2017.
 */

@Singleton
@Component(modules = {ApplicationModule.class})
public interface ApplicationComponent {

    void inject(HomePresenter presenter);

    void inject(FlyOnTimeApplication application);

}

File: app/src/main/java/org/flyontime/android/dagger/ApplicationComponent.kt
package org.flyontime.android.dagger

import dagger.Component
import org.flyontime.android.FlyOnTimeApplication
import org.flyontime.android.ui.presenter.FetchEmailsPresenter
import org.flyontime.android.ui.presenter.HomePresenter
import javax.inject.Singleton

/**
 * Created by JW on 27.04.2017.
 */

@Singleton
@Component(modules = arrayOf(ApplicationModule::class))
interface ApplicationComponent {

    fun inject(presenter: HomePresenter)

    fun inject(presenter: FetchEmailsPresenter)

    fun inject(application: FlyOnTimeApplication)

}
